st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -53,8 +53,12 @@ def __init__(self, chainspec, chainname):
         LOG.debug('Using %s as default for %s.', self.actionchain.default, self.chainname)
         if not self.actionchain.default:
             raise Exception('Failed to find default node in %s.' % (self.chainname))
-        # finalize the vars and save them around to be used at execution time.
-        self.vars = self._get_rendered_vars(self.actionchain.vars) if self.actionchain.vars else {}
+        self.vars = {}
+
+    def init_vars(self, action_parameters):
+        if self.actionchain.vars:
+            self.vars = self._get_rendered_vars(self.actionchain.vars,
```

Comment:
This change will work but I am wondering why we aren't letting this code https://github.com/StackStorm/st2/blob/enhanced_chain_vars/st2actions/st2actions/runners/actionchainrunner.py#L330 render vars before evaluating action_params?